src/main/java/duke/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,60 @@
+package duke.command;
+
+import duke.ui.Ui;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.TaskList;
+/**
+ * It is a command object extends from Command for the Duke program.
+ * When the parser calls it, it will receive the requests from the users
+ * during the running of the program and starts add new deadlines to the task list.
+ */
+public class AddDeadlineCommand extends Command{
+
+    /**
+     * Constructor for AddDeadlineCommand object
+     *
+     * @param userMessage The message that the user inputs for further execution.
+     */
+    public AddDeadlineCommand(String userMessage){
+        super(userMessage);
+    }
+
+    /**
+     * The execution after parsing, it will add a deadline object into the tasks.
+     * If the input is not correct, it will raise an exception.
+     *
+     * @param taskList The current taskList in the program.
+     * @param ui The current ui in the program.
+     * @throws DukeException if there are some cases such as no deadline time specified, no deadline task name, then
+     * it will raise the DukeException.
+     */
+    public void execute(TaskList taskList, Ui ui) throws DukeException {
+        StringBuilder builder = new StringBuilder();
```

Comment:
Just curious, why do you choose to use StringBuilder here?

Reply:
Because I read from materials that if you want to combine a lot of pieces of strings into a string then it's better to use the StringBuilder, because it will be faster than using "+" to combine strings into one, especially when you have a lot of pieces of strings.

Reply:
Ah I see, thanks for sharing!